server/controllers/entities/lib/get_entities_by_isbns.ts

```diff
@@ -3,34 +3,44 @@ import { getInvEntitiesByIsbns } from '#controllers/entities/lib/entities'
 import type { EntitiesGetterParams } from '#controllers/entities/lib/get_entities_by_uris'
 import { prefixifyIsbn } from '#controllers/entities/lib/prefix'
 import { enrichAndGetEditionEntityFromIsbn } from '#data/dataseed/enrich_and_get_edition_entity_from_isbn'
+import { getWdEntitiesByIsbns } from '#data/wikidata/get_wd_entities_by_isbns'
 import { parseIsbn } from '#lib/isbn/parse'
 import { getFirstClaimValue } from '#models/entity'
-import type { EntityUri, InvEntity, Isbn, IsbnEntityUri, RedirectFromTo, SerializedEntity } from '#types/entity'
+import type { IsbnData } from '#server/types/common'
+import type { EntityUri, ExpandedSerializedEntity, InvEntity, Isbn, IsbnEntityUri, RedirectFromTo, SerializedEntity } from '#types/entity'
 import { formatEditionEntity } from './format_edition_entity.js'
 
 export type Redirect = Record<EntityUri, EntityUri>
 
 export interface EntitiesResults {
-  entities: SerializedEntity[]
+  // This should ideally be corrected to `SerializedEntity[] | ExpandedSerializedEntity[]` but that causes a type error
+  entities: (SerializedEntity | ExpandedSerializedEntity)[]
   notFound?: IsbnEntityUri[]
 }
 
 export async function getEntitiesByIsbns (rawIsbns: Isbn[], params: EntitiesGetterParams = {}) {
-  const [ isbns, redirections ] = getRedirections(rawIsbns)
+  const { isbns13h: isbns, redirections, parsedIsbnsData } = getIsbnsData(rawIsbns)
   const { autocreate, refresh } = params
+  const { entities: serializedWdEntities } = await getWdEntitiesByIsbns(parsedIsbnsData, params)
```

Comment:
Its quite a piece this `getEntitiesByIsbns` now, but if i get it well, there is a case that seem problematic to me:
when an inv entity with a given isbn, let's say an edition with claims carefully crafted over the years by inv users, also have a wikidata item with the same isbn, whats happening ? The wikidata edition will be returned, ignoring the inv entity (right?) 
If this is the case, that means some editions data will be displayed differently (ie. merging this as is will change some titles to be a wd item title, instead of the inv edition now). Wouldn't it be safer (and more respectful of inv user contributions) to give inv editions priority over wd editions?

Reply:
That's why there is the after-merge TODO to "reconcile existing inv and wd edition entities", which would include transferring the carefully crafted claims to Wikidata. After that, the inv entity would become a redirection, or, if it has a book cover (`invp:P2`), a local entity layer.

Good point on the title: maybe `wdt:P1476` should be a required property to consider a Wikidata edition as acceptable, just like `wdt:P629`.

Reply:
A more precise illustration in the same shape as my previous comment:
As a long term inv user, I crafted my series on inventaire some times ago with the great `/cleanup` tool, formatted the titles etc. Then, after this PR merge, my inventory will look like having different title formats for the same series I crafted.

About moving inv editions to wikidata, what if after the inv entity is becoming a redirection, then the wd edition becomes a work?

It feels Inventaire is going all the way from not accepting any wd edition to give priority to all wd editions. By leaving priority to inventaire community in this PR, we would ensure some minimal moves (roadmap is tight), and then switch to wikidata editions fully at the time the bulk redirection is implemented?

Reply:
>  Then, after this PR merge, my inventory will look like having different title formats for the same series I crafted.

It should not be the case, as inv title conventions are based on wd conventions.
Additionally, it will most likely be the case that your series do not have a matching edition in Wikidata yet: in the latest inv dump, there were 1030831 editions with an ISBN, while there are only [80327 editions with an ISBN in Wikidata (at least that we would consider valid)](https://query.wikidata.org/#SELECT%20%3Fedition%20%3Fisbn13h%20%3Fisbn10h%20WHERE%20%7B%0A%20%20%3Fedition%20wdt%3AP31%20wd%3AQ3331189%20.%0A%20%20%7B%20%3Fedition%20wdt%3AP212%20%3Fisbn13h%20.%20%7D%20UNION%20%7B%20%3Fedition%20wdt%3AP957%20%3Fisbn10h%20.%20%7D%0A%20%20FILTER%20EXISTS%20%7B%0A%20%20%20%20%3Fedition%20wdt%3AP629%20%3Fwork%20.%0A%20%20%20%20%3Fedition%20wdt%3AP1476%20%3Ftitle%20.%0A%20%20%7D%0A%7D). Furthermore, the [overlap with the inv editions with an ISBN, still based on this latest inv dump, concerns only 12005 editions](https://dumps.inventaire.io/tmp/wd_inv_editions_with_isbn_overlap.ndjson). I would expect a small minority of those to be conflicting.

> what if after the inv entity is becoming a redirection, then the wd edition becomes a work?

That's unfortunately a very legitimate concern from our experience, but my hope is that only accepting editions that already have an associated work (`wdt:P629`) would make this occurrence quite rare: most cases of such a type change seem to occur for isolated "book" items.

Reply:
@jum-s  I inverted the priorities in 1532b2dce: existing local entities should now be found first, until they are merged with the remote entity.

Reply:
Could it be rebased (there are some conflicts) so that couch doesnt have to reindex all entities?

Reply:
@jum-s rebased